cpp/examples/tutorial_examples/file_access_example.cc

```diff
@@ -180,8 +180,9 @@ arrow::Status RunMain() {
   // (Doc section: Parquet OpenFile)
   // Note that Parquet's OpenFile() takes the reader by reference, rather than returning
   // a reader.
-  PARQUET_THROW_NOT_OK(
-      parquet::arrow::OpenFile(infile, arrow::default_memory_pool(), &reader));
+  auto result = parquet::arrow::OpenFile(infile, arrow::default_memory_pool());
+  PARQUET_THROW_NOT_OK(result.status());
+  reader = std::move(result.ValueOrDie());
```

Comment:
We have a convenient macro for this:

```suggestion
  PARQUET_ASSIGN_OR_THROW(reader, parquet::arrow::OpenFile(infile, arrow::default_memory_pool()));
```

For example: https://github.com/apache/arrow/pull/44785/files#diff-d3ed5886703305596174873a3a9a676cf8fd049707539ce3eed6b0a2551a674dL88-R89